trin-core/src/jsonrpc/utils.rs

```diff
@@ -0,0 +1,52 @@
+use crate::portalnet::Enr;
+use discv5::{
+    enr::NodeId,
+    kbucket::{ConnectionState, NodeStatus},
+};
+use ethereum_types::U256;
+use serde_json::{json, Value};
+use std::collections::HashMap;
+
+type NodeMap = HashMap<String, String>;
+type NodeTuple = (NodeId, Enr, NodeStatus, U256);
+
+/// Converts the output of the Overlay's bucket_entries method to a JSON Value
+pub fn bucket_entries_to_json(bucket_entries: HashMap<usize, Vec<NodeTuple>>) -> Value {
+    let mut node_count: u16 = 0;
+    let mut connected_count: u16 = 0;
+    let buckets_indexed: HashMap<usize, Vec<NodeMap>> = bucket_entries
+        .into_iter()
+        .map(|(bucket_index, bucket)| {
+            (
+                bucket_index,
+                bucket
+                    .iter()
+                    .map(|(node_id, enr, node_status, data_radius)| {
+                        node_count += 1;
+                        if node_status.state == ConnectionState::Connected {
+                            connected_count += 1
+                        }
+                        let mut map = HashMap::new();
+                        map.insert(
+                            "node_id".to_owned(),
+                            format!("0x{}", hex::encode(node_id.raw())),
+                        );
+                        map.insert("enr".to_owned(), enr.to_base64());
+                        map.insert("status".to_owned(), format!("{:?}", node_status.state));
+                        map.insert("radius".to_owned(), format!("{}", data_radius));
+                        map
+                    })
+                    .collect(),
+            )
+        })
+        .collect();
+
+    json!(
+        {
+            "buckets": buckets_indexed,
```

Comment:
I see that the example has buckets sorted by index, and I agree that it's something we want. I didn't see any explicit sorting here. On a quick search, I wasn't able to confirm that serde's `json!` uses a serialization that orders by keys, either.

I found some references to `#[serde(serialize_with="ordered_map")]` online, if that helps.

Reply:
Looks like that approach was deprecated, and now you just choose map ordering behavior by enabling a feature in `Cargo.toml`.